### PR TITLE
Fixed the left mouse click for mesh editing not working with the new shortcut improvements.

### PR DESCRIPTION
**Pull request by claudiaantoun**
### Purpose of this PR

This PR fixes an issue where it was no longer possible to do mesh editing or enable view tools after the new shortcut improvements were added to the editor.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Fixed
 
+- Fixed a bug where it was no longer possible to edit meshes or enable view tools after the new shortcut improvements were added to the editor.
```
Comment (lopezt-unity) [inline CHANGELOG.md:12]: This should be in a new ## [Unreleased] section and not in 5.2.0

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 2]
```diff
@@ -567,7 +567,15 @@ void OnSceneGUI(SceneView sceneView)
 
         internal void HandleMouseEvent(SceneView sceneView, int controlID)
         {
-            if(m_CurrentEvent.type == EventType.MouseDown && HandleUtility.nearestControl == controlID)
+#if UNITY_2023_1_OR_NEWER
```
Comment (lopezt-unity) [inline Editor/EditorCore/ProBuilderEditor.cs:570]: Should this also be done for the context menu PR ?

**Comment by lopezt-unity** [comment]
I just created a `v5.2/master` branch for next 5.2.X release, this PR should target this new branch rather than `master` I think :)

File: CHANGELOG.md (hunk 1 above)
Comment (claudiaantoun) [reply CHANGELOG.md:12]: Oops! Fixed in [cb02fe5](https://github.com/Unity-Technologies/com.unity.probuilder/pull/487/commits/cb02fe5a2792e9639cd384b686adfef47d538d04)

**Review by lopezt-unity — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Fixed
+
+- Fixed a bug where it was no longer possible to edit meshes or enable view tools after the new shortcut improvements were added to the editor.
```
Comment (JonMUnity) [inline CHANGELOG.md:12]:
```suggestion
- Fixed a bug that prevented editing meshes or enabling view tools after the new shortcut improvements were added to the Editor.
```

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 4]
```diff
@@ -567,7 +567,18 @@ void OnSceneGUI(SceneView sceneView)
 
         internal void HandleMouseEvent(SceneView sceneView, int controlID)
         {
-            if(m_CurrentEvent.type == EventType.MouseDown && HandleUtility.nearestControl == controlID)
+            if (Tools.viewToolActive || Tools.current == Tool.View || (Event.current.modifiers & EventModifiers.Alt) == EventModifiers.Alt)
```
Comment (karl-) [inline Editor/EditorCore/ProBuilderEditor.cs:570]: Everything after `Tools.viewToolActive` should be redundant, is that the case?

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 5]
```diff
@@ -567,7 +567,18 @@ void OnSceneGUI(SceneView sceneView)
 
         internal void HandleMouseEvent(SceneView sceneView, int controlID)
         {
-            if(m_CurrentEvent.type == EventType.MouseDown && HandleUtility.nearestControl == controlID)
+            if ((Event.current.modifiers & EventModifiers.Alt) == EventModifiers.Alt)
```
Comment (karl-) [inline Editor/EditorCore/ProBuilderEditor.cs:570]: There is some code in this function that handles exiting drag events. What happens if during a drag the `alt` key is held and a mouse button up event occurs? Does the drag exit successfully?

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 6]
```diff
@@ -567,7 +567,18 @@ void OnSceneGUI(SceneView sceneView)
...
+                return;
+
+#if UNITY_2023_1_OR_NEWER
+            bool isLeftMouseClick = m_CurrentEvent.type == EventType.MouseDown && m_CurrentEvent.keyCode == KeyCode.Mouse0 && HandleUtility.nearestControl == controlID;
+#elif UNITY_2022_3
+            KeyCode keyCode = m_CurrentEvent.isMouse ? m_CurrentEvent.button + KeyCode.Mouse0 : m_CurrentEvent.keyCode;
+            bool isLeftMouseClick = m_CurrentEvent.type == EventType.MouseDown && keyCode == KeyCode.Mouse0 && HandleUtility.nearestControl == controlID;
+#else
+            bool isLeftMouseClick = m_CurrentEvent.type == EventType.MouseDown && HandleUtility.nearestControl == controlID;
```
Comment (karl-) [inline Editor/EditorCore/ProBuilderEditor.cs:579]: This case isn't testing for left button, is that intentional?

Comment (claudiaantoun) [reply Editor/EditorCore/ProBuilderEditor.cs]: I thought I'd leave the code as it was before the shortcut changes, but I realize that it's not necessary. Fixed in [a7935ea](https://github.com/Unity-Technologies/com.unity.probuilder/pull/487/commits/a7935ea69bab0ff81f2bef19eebaeab31d355a32)

File: Editor/EditorCore/ProBuilderEditor.cs (hunk 5 above)
Comment (claudiaantoun) [reply Editor/EditorCore/ProBuilderEditor.cs]: I've added, in [a7935ea](https://github.com/Unity-Technologies/com.unity.probuilder/pull/487/commits/a7935ea69bab0ff81f2bef19eebaeab31d355a32), a check for `m_IsDragging` in that `if` statement to make sure that the rect selection is not interrupted if the `alt` key is pressed in the middle of the rect selection.

**Review by JonMUnity — APPROVED** [review]

**Review by lopezt-unity — APPROVED** [review]
LGTM, tested with Claudia

**Review by karl- — APPROVED** [review]

**Pull request merged**